python/pyarrow/_table.pyx

```diff
@@ -321,8 +322,11 @@ cdef tuple _dataframe_to_arrays(df, bint timestamps_to_ms, Schema schema):
     cdef:
```

Comment:
This function is getting "chubby" enough that we should probably move it to a pandas utility module in pure Python.